### ISSUE TITLE
`reportDeprecated` false negative on deprecated class assigned to `TypeForm`

### ISSUE THREAD

**Issue by DetachHead** [DetachHead/basedpyright#1720]
### Description

Code sample in [basedpyright playground](https://basedpyright.com/?pyrightVersion=1.37.2&typeCheckingMode=all&enableExperimentalFeatures=true&reportMissingModuleSource=false&code=GYJw9gtgBA7ghiAdgS0QcwM5WRADmEAFygBMBTXEMgYzkLJIFgAoF0SKQgT11TQH0yAD3qIMyMGOx4CxACo8yAMQIQW65gAFylGnQYAKAERwMJYEYCULagBtTWFWABcUAHQeNLU%2BdcLcyqoA2mAARgBWNIQAulAAvFBOUFAAxFCIYKQUVLSEEoiwCCjoQA)

```python
from warnings import deprecated

from typing_extensions import TypeForm


@deprecated("asdf")
class Foo: ...


asdf: TypeForm[object] = Foo  # no deprecation warning
```
